content/build/bake/inheritance.md

```diff
@@ -0,0 +1,165 @@
+  labels = {
+    "org.opencontainers.image.source" = "https://github.com/username/myapp"
+    "org.opencontainers.image.author" = "moby.whale@example.com"
+  }
+}
+```
+
+You can create a new target that uses the same build configuration, but with
+slightly different attributes for a production build. In this example, the
+`app-release` target inherits the `app-dev` target, but overrides the `tags`
+attribute and adds a new `platforms` attribute:
+
+```hcl
+target "app-release" {
+  inherits = ["app-dev"]
+  tags = ["docker.io/username/myapp:latest"]
+  platforms = ["linux/amd64", "linux/arm64"]
+}
+```
+
+## Common reusable targets
+
+One common inheritance pattern is to define a common target that contains
+shared attributes for all or many of the build targets in the project. For
+example, the following `_common` target defines a common set of build
+arguments:
+
+```hcl
+target "_common" {
+  args = {
+    GO_VERSION = "{{% param example_go_version %}}"
+    BUILDKIT_CONTEXT_KEEP_GIT_DIR = 1
+  }
+}
+```
+
+You can then inherit the `_common` target in other targets to apply the shared
+attributes:
+
+```hcl
+target "lint" {
+  inherits = ["_common"]
+  dockerfile = "./dockerfiles/lint.Dockerfile"
+  output = ["type=cacheonly"]
+}
+
+target "docs" {
+  inherits = ["_common"]
+  dockerfile = "./dockerfiles/docs.Dockerfile"
+  output = ["./docs/reference"]
+}
+
+target "test" {
+  inherits = ["_common"]
+  target = "test-output"
+  output = ["./test"]
+}
+
+target "binaries" {
+  inherits = ["_common"]
+  target = "binaries"
+  output = ["./build"]
+  platforms = ["local"]
+}
+```
+
+## Overriding inherited attributes
+
+When a target inherits another target, it can override any of the inherited
+attributes. For example, the following target overrides the `args` attribute
+from the inherited target:
+
+```hcl
+target "app-dev" {
+  inherits = ["_common"]
+  args = {
+    GO_VERSION = "1.17"
+  }
+  tags = ["docker.io/username/myapp:dev"]
+}
+```
+
+The `GO_VERSION` argument in `app-release` is set to `1.17`, overriding the
+`GO_VERSION` argument from the `app-dev` target.
+
+For more information about overriding attributes, see the [Overriding
+configurations](./overrides.md) page.
+
+## Inherit from multiple targets
+
+The `inherits` attribute is a list, meaning you can reuse attributes from
+multiple other targets. In the following example, the app-release target reuses
+attributes from both the `app-dev` and `_common` targets.
+
+```hcl
+target "_common" {
+  args = {
+    GO_VERSION = "{{% param example_go_version %}}"
+    BUILDKIT_CONTEXT_KEEP_GIT_DIR = 1
+  }
+}
+
+target "app-dev" {
+  inherits = ["_common"]
+  args = {
+    BUILDKIT_CONTEXT_KEEP_GIT_DIR = 0
+  }
+  tags = ["docker.io/username/myapp:dev"]
+  labels = {
+    "org.opencontainers.image.source" = "https://github.com/username/myapp"
+    "org.opencontainers.image.author" = "moby.whale@example.com"
+  }
+}
+
+target "app-release" {
+  inherits = ["app-dev", "_common"]
```

Comment:
Don't think this is example is meaningful as this would have the same effect as using:

```hcl
  inherits = ["app-dev"]
```

Reply:
Ah nevermind, just figure out about `BUILDKIT_CONTEXT_KEEP_GIT_DIR` build-arg.

Reply:
yeah, bit of a tongue twister but just wanted to show the inheritance order.